### ISSUE TITLE
Installation Fork CMS 4.5.4 on Windows Server 2016 / IIS 10 / PHP 7.1.7 fails

### ISSUE THREAD

**Issue by AaTTaxX** [forkcms/forkcms#2132]
Hi, I am trying to run a Fork CMS installation on: 

- Windows Server 2016 / 
- IIS 10.0.14393.0 
- PHP version 7.1.7 x64 NTS
- Fork CMS 4.5.4

Unfortunately, the first steps have already failed.

My steps:

    1. Fork CMS 4.5.4 downloaded

    2. Fork CMS unpacked under the path: C:\inetpub\wwwroot\

    3. New website created in IIS -> Site name: fork

    4. Application pool: DefaultAppPool

    5. Physical path: C:\inetpub\wwwroot\
![fork](https://user-images.githubusercontent.com/30074120/28065078-5ca754d0-6637-11e7-8858-94ee3497c4aa.png)

    6. Explorer the physical path via IIS -> (**works**)
![fork2](https://user-images.githubusercontent.com/30074120/28065096-6cf399e8-6637-11e7-9671-821b478bee5c.png)

    7. Open Firefox and go to "http://localhost/", Firefox is trying to load "http://localhost/install" but there is no location like this

    8. Receive error: HTTP Error 404.0 - Not Found
![fork3](https://user-images.githubusercontent.com/30074120/28065142-9ecaa902-6637-11e7-9d44-3f110ed8c7b7.png)

Detailed Error Information:
Module: IIS Web Core
Notification: MapRequestHandler
Handler: StaticFile
Error Code: 0x80070002
Requested URL: http://localhost:80/install
Physical Path: C:\inetpub\wwwroot\install
Logon Method: Anonymous
Logon User: Anonymous

**Comment by carakas** [comment]
does htaccess work on that server? if not you need to port that configuration to what works on windows

**Comment by AaTTaxX** [comment]
have converted the .htaccess to web.config but still get error but another one "Corrupted Content Error"
Here my web.config:
`<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <system.webServer>
        <rewrite>
            <rules>
                <rule name="Imported Rule 1">
                    <match url=".*" ignoreCase="false" />
                    <conditions logicalGrouping="MatchAll">
                        <add input="{REQUEST_METHOD}" pattern="^TRACE" ignoreCase="false" />
                    </conditions>
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 2" stopProcessing="true">
                    <match url="^index\.php$" ignoreCase="false" />
                    <action type="None" />
                </rule>
                <rule name="Imported Rule 3" stopProcessing="true">
                    <match url="src/Backend/Core/Js/ckfinder/core/connector/php/connector\.php" ignoreCase="false" />
                    <action type="None" />
                </rule>
                <rule name="Imported Rule 4">
                    <match url=".*\.php" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 5">
                    <match url="\.git" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 6">
                    <match url="vendor/.*" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 7">
                    <match url="bin/.*" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 8">
                    <match url="src/Console/.*" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 9">
                    <match url=".*\.gitignore" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 10">
                    <match url="\.editorconfig" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 11">
                    <match url="\.travis.yml" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 12">
                    <match url="bower\.json" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 13">
                    <match url="composer\.json" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 14">
                    <match url="composer\.lock" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 15">
                    <match url="phpunit\.xml\.dist" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 16">
                    <match url=".*\.md" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 17">
                    <match url="app/logs" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 18">
                    <match url="app/config" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 19">
                    <match url="src/Frontend/Cache/CompiledTemplates" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 20">
                    <match url="src/Frontend/Cache/Search" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 21">
                    <match url="src/Backend/Cache/CompiledTemplates" ignoreCase="false" />
                    <action type="CustomResponse" statusCode="403" statusReason="Forbidden" statusDescription="Forbidden" />
                </rule>
                <rule name="Imported Rule 22" stopProcessing="true">
                    <match url="^(.*)$" ignoreCase="false" />
                    <conditions logicalGrouping="MatchAll">
                        <add input="{SERVER_PORT}" pattern="80" ignoreCase="false" />
                        <add input="{HTTP_HOST}" pattern=".*\.dev" negate="true" />
                    </conditions>
                    <action type="Redirect" url="https://www.&lt;domain>.be/{R:1}" redirectType="Permanent" />
                </rule>
                <rule name="Imported Rule 23" stopProcessing="true">
                    <match url="^(.*)$" ignoreCase="false" />
                    <conditions logicalGrouping="MatchAll">
                        <add input="{HTTP_HOST}" pattern="^www\.&lt;domain>\.be" negate="true" />
                        <add input="{HTTP_HOST}" pattern=".*\.dev" negate="true" />
                    </conditions>
                    <action type="Redirect" url="http://www.&lt;domain>.be/{R:1}" redirectType="Permanent" />
                </rule>
                <rule name="Imported Rule 24" stopProcessing="true">
                    <match url="." />
                    <conditions logicalGrouping="MatchAll">
                        <add input="{URL}" pattern="^/(src|src\/)$" ignoreCase="false" />
                    </conditions>
                    <action type="Rewrite" url="/index.php" />
                </rule>
                <rule name="Imported Rule 25" stopProcessing="true">
                    <match url="." />
                    <conditions logicalGrouping="MatchAll">
                        <add input="{URL}" pattern="^$" ignoreCase="false" negate="true" />
                        <add input="{REQUEST_FILENAME}" matchType="IsFile" ignoreCase="false" negate="true" />
                        <add input="{REQUEST_FILENAME}" matchType="IsDirectory" ignoreCase="false" negate="true" />
                    </conditions>
                    <action type="Rewrite" url="index.php" />
                </rule>
            </rules>
        </rewrite>
    </system.webServer>
</configuration>
`

**Comment by carakas** [comment]
I don't know enough of windows servers to be able to help you with that but I'll post it in the slack channel, feel free to join it as well
https://fork-cms.herokuapp.com/

**Comment by AaTTaxX** [comment]
No windows lovers there, with the same problem?

**Comment by carakas** [comment]
It would seem so, I'm sorry

**Comment by jeroendesloovere** [comment]
Hi @AaTTaxX,  did you try to build up the htaccess from "required to run" to "production ready"?

F.e., can you start with:

```
<IfModule mod_rewrite.c>
	# set environment variable to let PHP know that urls are being rewritten
	#RewriteRule .* - [E=MOD_REWRITE:1]

	# Protect against cross site tracing https://www.owasp.org/index.php/Cross_Site_Tracing
	#RewriteCond %{REQUEST_METHOD} ^TRACE
	#RewriteRule .* - [F]

	# nice urls
	RewriteEngine On
	RewriteBase /

        # allow the ck connector
        RewriteRule ^index\.php$ - [L]
        RewriteRule src/Backend/Core/Js/ckfinder/core/connector/php/connector\.php - [L]

	# forbidden folders
        RewriteRule .*\.php - [F]
	RewriteRule \.git - [F]
	RewriteRule vendor/.* - [F]
	RewriteRule bin/.* - [F]
	RewriteRule src/Console/.* - [F]
	RewriteRule .*\.gitignore - [F]
	RewriteRule \.editorconfig - [F]
	RewriteRule \.travis.yml - [F]
	RewriteRule bower\.json - [F]
	RewriteRule composer\.json - [F]
	RewriteRule composer\.lock - [F]
	RewriteRule phpunit\.xml\.dist - [F]
	RewriteRule .*\.md - [F]
	RewriteRule app/logs - [F]
	RewriteRule app/config - [F]
	RewriteRule src/Frontend/Cache/CompiledTemplates - [F]
	RewriteRule src/Frontend/Cache/Search - [F]
	RewriteRule src/Backend/Cache/CompiledTemplates - [F]

	# redirect all trafic to https
	# RewriteCond %{SERVER_PORT} 80
	# RewriteCond %{HTTP_HOST} !.*\.dev [NC]
	# RewriteRule ^(.*)$ https://www.<domain>.be/$1 [R=301,L]

	# one url to rule them all
	# RewriteCond %{HTTP_HOST} !^www\.<domain>\.be [NC]
	# RewriteCond %{HTTP_HOST} !.*\.dev [NC]
	# RewriteRule ^(.*)$ http://www.<domain>.be/$1 [R=301,L]

	# src dir should pass via the front controller
	RewriteCond %{REQUEST_URI} ^/(src|src\/)$
	RewriteRule . /index.php [NC,L]

	# handle urls
	RewriteCond %{REQUEST_URI} !^$
	RewriteCond %{REQUEST_FILENAME} !-f
	RewriteCond %{REQUEST_FILENAME} !-d
	RewriteRule . index.php [NC,L]
</IfModule>
```

**Comment by jeroendesloovere** [comment]
@AaTTaxX, it's like rare Pokémons, those Windows lovers.
Rarely you see one and when you try to catch him then... Your windows crashes...
![windows-end-program](https://user-images.githubusercontent.com/588616/28324568-06888f7e-6bdc-11e7-9516-2254371cf06e.png)

**Comment by AaTTaxX** [comment]
Hi Jeroen,

.htaccess is not working with IIS (only with 3rd party tools).

I have tested with a Helicon tool that provides support for Apache .htaccess configuration for IIS and it seems to work, but I do not really want to use a 3rd party tool.

Saw this **nice** screen for the first time on a Windows Server:
![installation](https://user-images.githubusercontent.com/30074120/28355988-a013e370-6c66-11e7-8a97-e6b78aa17332.png)

I have tried to convert the ".htaccess" to "web.config" but at two points I get an error.

**1st point:**
![1st point](https://user-images.githubusercontent.com/30074120/28356009-b84a3188-6c66-11e7-80b5-6990a669712d.png)

**2nd point:**
![2nd point](https://user-images.githubusercontent.com/30074120/28356010-bb39cd9a-6c66-11e7-83b3-2306db63b664.png)

It fails only on the "web.config" file, which can not be converted 1:1.

Jeroen, rare Pokemon are epic and legendary as well as Windows lovers but with the error message you are probably right....
